### PR TITLE
Kaizen: configure page metadata based on article content

### PR DESCRIPTION
**Pull request by glsignal**
Hopefully this improves the link preview situation a little bit!

### CODE REVIEW

**Comment by UlianaMunich** [comment]
Thanks a lot, @glsignal ! 

*Not urgent:
Twitter Icon can't be loaded. 

Screenshot 2023-09-15 at 19 49 41
Screenshot 2023-09-15 at 19 49 36

**Comment by glsignal** [comment]
Thanks @UlianaMunich - I wasn't able to trigger any load failures for the icons, I wonder if it's somehow pre-existing in the project?

File: src/layouts/BlogPost.astro [hunk 1]
```diff
@@ -2,40 +2,30 @@
...
-  thumbnailSrc,
-} = Astro.props;
+const post = Astro.props;
 
-const authorsData = await getEntries(authors);
+const authorsData = await getEntries(post.authors);
 ---
 
-<BaseLayout>
+<BaseLayout post={post}>
```
Comment (dmprav) [inline src/layouts/BlogPost.astro:15]: Why do you need to pass it here?

**Review by dmprav — APPROVED** [review]
Looks great, thanks a lot! ✨ 

I couldn't reproduce the icon issue.

File: src/layouts/BlogPost.astro (hunk 1 above)
Comment (glsignal) [reply src/layouts/BlogPost.astro:15]: I don't _fully_ remember to be honest, though I vaguely remember running into some issues with props not always being defined/accessible from within the base layout when it's nested in this way (it doesn't seem to inherit access to the props automatically).
This _might_ change if we start specifying the layout in the yaml frontmatter, but I didn't really explore that too much

**Pull request merged**